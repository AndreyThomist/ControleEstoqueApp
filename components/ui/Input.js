import React, { useState } from 'react'
import { View,Text, StyleSheet, TextInput } from 'react-native'
const Input = (props) => {
    return (
        <View style={{ ...styles.container,...props.estilo }}>
            <Text>{props.label}</Text>
            <View style={styles.formContainer}>
            <TextInput
                {...props}
            />
            </View>
            <View style={styles.errors}>
                     {props.errorMessage && props.touched ? <Text style={{color:'red'}}>{props.errorMessage}</Text> : null}
            </View>
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        marginVertical:10,
        width:'85%'
    },
    formContainer:{
        borderBottomColor: '#000000',
        borderBottomWidth: 1,

    },
    errors:{
        marginVertical:5
    }
})

export default Input;