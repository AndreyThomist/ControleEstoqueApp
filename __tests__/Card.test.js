import React from 'react'
import renderer from 'react-test-renderer'
import Card from '../components/ui/Card'
import Input from '../components/ui/Input'

/* Componente Auxiliar estÃ¡ renderizando*/
it('renders correctly',() => {
    const tree = renderer.create(<Card><Input /></Card>).toJSON();
    expect(tree).toMatchSnapshot();
})
