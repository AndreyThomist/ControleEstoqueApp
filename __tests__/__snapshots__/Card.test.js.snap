// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "elevation": 5,
    }
  }
>
  <View
    style={
      Object {
        "marginVertical": 10,
        "width": "85%",
      }
    }
  >
    <Text />
    <View
      style={
        Object {
          "borderBottomColor": "#000000",
          "borderBottomWidth": 1,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        rejectResponderTermination={true}
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      style={
        Object {
          "marginVertical": 5,
        }
      }
    />
  </View>
</View>
`;
